src/Http/Client/Adapter/Curl.php

```diff
@@ -149,7 +149,7 @@ public function buildOptions(RequestInterface $request, array $options): array
             $out[CURLOPT_PROXY] = $options['proxy']['proxy'];
         }
         if (isset($options['proxy']['username'])) {
-            $password = !empty($options['proxy']['password']) ? $options['proxy']['password'] : '';
+            $password = empty($options['proxy']['password']) ? '' : $options['proxy']['password'];
```

Comment:
```suggestion
            $password = $options['proxy']['password'] ?? '';
```

Reply:
Could it be empty string though?

Reply:
The default value is any empty string anyway, so won't matter if array value is an empty string, the result is the same.